js/main.js

```diff
@@ -365,43 +303,113 @@ function updateLegendDataSource(dataSource) {
     $("#legend #dataSource").html('<a href="' + url + '">' + title + '</a>');
 }
 
+
+/*
+ * Convert a city ID to a UI label.
+ */
+function cityIdToLabel(s) {
+    return s.replace(/(^|\s|\-)([a-zA-ZäöüÄÖÜß])/g, function (x) {
+        return x.toUpperCase();
+    });
+}
+
+
 /*
- * Initialize application when market data is loaded.
+ * Set the current city.
+ *
+ * `city` is the city ID.
  */
-function init(json, cityName) {
-    positionMap(json.metadata.map_initialization);
-    updateLegendDataSource(json.metadata.data_source);
-    initMarkers(json);
-    initControls();
-    map.addLayer(unclassifiedGroup);
-    map.addLayer(nowGroup);
-    updateLayers();
-    updateDocumentTitle(cityName);
-    updateUrlHash(cityName);
+function setCity(city) {
+    var filename = 'cities/' + city + '.json';
+    $.getJSON(filename, function(json) {
+        positionMap(json.metadata.map_initialization);
+        updateLegendDataSource(json.metadata.data_source);
+        initMarkers(json);
+        initControls();
+        map.addLayer(unclassifiedGroup);
+        map.addLayer(nowGroup);
+        updateLayers();
+        updateUrlHash(city);
+        document.title = 'Wo ist Markt in ' + cityIdToLabel(city) + '?';
+        $('#dropDownCitySelection').val(city).trigger('change', true);
+    }).fail(function() {
+        console.log('Failure loading "' + filename + '".');
+        if (city !== DEFAULT_CITY) {
+            console.log('Loading default city "' + DEFAULT_CITY +
+                        '" instead.');
+            setCity(DEFAULT_CITY);
+        }
+    });
 }
 
+
 /*
- * Forces reloading when URI hash changed.
+ * Load the IDs of the available cities.
+ *
+ * Returns a jQuery `Deferred` object that resolves to an array of city IDs.
  */
+function loadCityIDs() {
+    var d = $.Deferred();
+    $.get(CITY_LIST_API_URL, function(result) {
+        var ids = [];
+        $.each(result, function(_, file) {
+            if (file.name.indexOf('.json', file.name.length - 5) !== -1) {
```

Comment:
Since we're going for a local list of cities anyway (#36) I won't change this. It's basically the code we had before, just factored out into a separate function.